### ISSUE TITLE
multiline-ternary "never" allows multiline ternaries

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#13368]
**Tell us about your environment**

* **ESLint Version:** v7.1.0
* **Node Version:** v12.14.0
* **npm Version:** v6.13.4

**What parser (default, Babel-ESLint, etc.) are you using?**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  parserOptions: {
      ecmaVersion: 2015
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG11bHRpbGluZS10ZXJuYXJ5OiBbXCJlcnJvclwiLCBcIm5ldmVyXCJdICovXG5cbmEgK1xuICAgIGIgPyBjICtcbiAgICAgICBkIDogZSArXG4gICAgICAgICAgIGY7Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo1LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

```js
/* eslint multiline-ternary: ["error", "never"] */

a +
    b ? c +
       d : e +
           f;  
```

```bash
eslint index.js
```

**What did you expect to happen?**

Given the documentation:

> "never" disallows newlines between the operands of a ternary expression (enforcing that the entire ternary expression is on one line).

This should be an error because the entire ternary expression is not on one line?

**What actually happened? Please include the actual, raw output from ESLint.**

No errors.

**Are you willing to submit a pull request to fix this bug?**

By #6751 it looks like this option should really enforce that the entire ternary expression is on one line, but it doesn't work like that from the start, so maybe we should just update the docs to reflect the actual behavior?

**Comment by kaicataldo** [comment]
@mdjermanovic Do you still think this is a bug? Or do you believe it falls under our new policy of not making changes to style rules?

**Comment by mdjermanovic** [comment]
I'm not sure, but anyway it would be a big change in the behavior of this rule.

I think we should just change the docs to match the actual behavior,

This should be removed:

> (enforcing that the entire ternary expression is on one line)

and some correct multiline examples should be added to clarify `"never"`.
